### PR TITLE
Add checking if given attributes are variant attributes in ProductVariantCreate mutation

### PR DESCRIPTION
**Pull request by L3str4nge**
1. If attributes in mutation input **are not** in `product_type.variant_attributes` -> raise Validation Error. This is fixing the situation where given attribute is not actually variant attribute.
2. If attributes in mutation input **are** in `product_type.variant_attributes` and `product_type.has_variants=False`-> raise Validation Error. This is fixing the situation where attributes input is not validated and raw data is passed to further logic.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2873 'str' object has no attribute 'input_type'](https://app.clickup.com/t/2549495/SALEOR-2873)

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -866,6 +866,21 @@ def clean_input(
...
+            )
+        ]
+
+        # Check if given attributes are variant attributes
+        for attr in data["attributes"]:
+            if attr["id"] not in variant_attributes_ids:
+                raise ValidationError(
+                    "Given attribute is not a variant attribute",
+                    ProductErrorCode.ATTRIBUTE_CANNOT_BE_ASSIGNED.value,
+                )
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/products.py:882]: IMO we can simplify this by creating sets for both lists and make difference. Also, I will suggest returning the invalid attributes as `params` in error like:
```
raise ValidationError(
      "Given attribute is not a variant attribute.",
      code=ProductErrorCode.ATTRIBUTE_CANNOT_BE_ASSIGNED.value,
      params={"attributes": invalid_attributes}
)
```
👮‍♀️ Remember about dot at the end of the error message.

File: saleor/graphql/product/tests/test_variant.py [hunk 2]
```diff
@@ -4674,3 +4674,73 @@ def test_query_product_variant_for_federation(api_client, variant, channel_USD):
...
+    response = staff_api_client.post_graphql(
+        VARIANT_CREATE_MUTATION,
+        variables={"productId": prod_id, "attrId": attr_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+
+    errors = content["data"]["productVariantCreate"]["errors"]
+    assert errors
+    assert errors[0]["message"] == "Given attribute is not a variant attribute"
```
Comment (IKarbowiak) [inline saleor/graphql/product/tests/test_variant.py:4720]: We usually do not check the error messages, we are checking `code` and `field` of an error. Checking messages is connected with fixing every test that checks this, even if a dot is added.

File: saleor/graphql/product/tests/test_variant.py [hunk 3]
```diff
@@ -4674,3 +4674,73 @@ def test_query_product_variant_for_federation(api_client, variant, channel_USD):
...
+    response = staff_api_client.post_graphql(
+        VARIANT_CREATE_MUTATION,
+        variables={"productId": prod_id, "attrId": attr_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+
+    errors = content["data"]["productVariantCreate"]["errors"]
+    assert errors
+    assert errors[0]["message"] == "Product type is not configurable"
```
Comment (IKarbowiak) [inline saleor/graphql/product/tests/test_variant.py:4746]: Same here.

File: saleor/graphql/product/mutations/products.py [hunk 4]
```diff
@@ -888,6 +903,11 @@ def clean_input(
                     )
             except ValidationError as exc:
                 raise ValidationError({"attributes": exc})
+        else:
+            raise ValidationError(
+                "Product type is not configurable",
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/products.py:908]: 👮‍♀️ 
```suggestion
                "Product type is not configurable.",
```

File: saleor/graphql/product/mutations/products.py [hunk 5]
```diff
@@ -866,6 +866,21 @@ def clean_input(
...
             )
 
+        variant_attributes_ids = [
+            graphene.Node.to_global_id("Attribute", attr_id)
+            for attr_id in list(
+                product_type.variant_attributes.all().values_list("pk", flat=True)
+            )
+        ]
+        attributes_ids = [attr["id"] for attr in data.get("attributes") or []]
+        invalid_attributes = set(attributes_ids) - set(variant_attributes_ids)
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/products.py:876]: You can use set comprehensions instead of changing lists into sets :)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/tests/fixtures.py [hunk 6]
```diff
@@ -930,6 +930,74 @@ def color_attribute(db):
...
+    return product
+
+
+@pytest.fixture
+def product_with_variant_attributes(
+    product_type_with_variant_attributes, non_default_category
+):
+    product = Product.objects.create(
+        name="Test product",
+        slug="test-product-11",
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:994]: IMO we should use a different slug for these products. In your implementation, we can't use both fixtures in one test. Same with the Product type.

Comment (L3str4nge) [reply saleor/tests/fixtures.py:994]: Good point :)

**Review by jakubkuc — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**